api/src/main/java/com/cloudera/livy/JobContext.java

```diff
@@ -59,6 +59,8 @@
   /** Stops the SparkStreaming context. */
   void stopStreamingCtx();
 
+  void stop();
```

Comment:
I haven't looked at the rest, but we shouldn't expose this. Clients should use `LivyClient.stop`.